src/vscode-dts/vscode.d.ts

```diff
@@ -16081,6 +16081,24 @@ declare module 'vscode' {
 		 * @returns A promise that resolves to the Debug Adapter Protocol breakpoint or `undefined`.
 		 */
 		getDebugProtocolBreakpoint(breakpoint: Breakpoint): Thenable<DebugProtocolBreakpoint | undefined>;
+
+		/**
+		 * Obtains information on a possible data breakpoint that could be set on an expression or variable.
+		 * This will fail if the corresponding capability `supportsDataBreakpoints` is not supported by this session.
+		 *
+		 * @param name The name of the variable's child to obtain data breakpoint information for. If `variablesReference` isn't specified, this can be an expression.
+		 * @param variablesReference Reference to the variable container if the data breakpoint is requested for a child of the container.
+		 */
+		getDataBreakpointInfo(name: string, variablesReference?: number): Thenable<DataBreakpointInfo | undefined>;
```

Comment:
Where would `variablesReference` come from? I don't think this is exposed in API elsewhere.

Reply:
You are correct that there is no API to directly retrieve the `variablesReference` at the moment. I made that choice to keep the API changes smaller but we could introduce such methods if we need to. The two main entry points I see that can currently be used to retrieve the reference is to:
- Register a debug adapter factory `vscode.debug.registerDebugAdapterTrackerFactory` and listen to the messages that are being sent and tracking the variable information since there is no direct access to the variables through the API.
- Call the debug adapter directly with `vscode.debug.activeDebugSession.customRequest` and use the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/specification) to access the variable information. For instance, in the C/C++ debugger, one could send an [evaluate request](https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Evaluate) to retrieve information about a variable if the variable name is known.

For now, I did not envision any particular support for the user in this regard but of course we could also look into extending the variable resolution area a bit more.